src/daria/analysis/segmentationcomparison.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
+
+import numpy as np
+
+
+def compare_segmentations(*segmentations, **kwargs) -> np.ndarray:
+    """
+    Comparison of segmentations.
+
+    Args:
+        segmentations (asterisk argument): Allows to provide an arbitraty
+                number of segmented images to be compared
+        optional keyword arguments (kwargs):
+            components (list): list of the different components that are
+                found in the segmented images. So far only two are allowed
+                to be provided.
+            gray_colors (np.ndarray): array of three different scales of
+                gray, one for each of the different combinations of
+                components in the segmentations.
+            colors (np.ndarray): Array of different colors that should
+                indicate uniqe components in each segmentation.
+            light_scaling (float): Indicate how much lighter the second
+                component should be scaled in its unique color.
+    """
+    # Define components
+    components: list = kwargs.pop("components", [1, 2])
+
+    # Define number of segmentations
+    N: int = len(segmentations)
+
+    # Define gray colors
+    gray_colors: np.ndarray = kwargs.pop(
+        "gray_colors",
+        np.array([[180, 180, 180], [220, 220, 220], [200, 200, 200]], dtype=np.uint8),
+    )
+    gray_base: np.ndarray = np.fix(gray_colors / N).astype(np.uint8)
+
+    # Define unique colors
+    colors: np.ndarray = kwargs.pop(
+        "colors",
+        np.array(
+            [
+                [[205, 0, 0]],
+                [[0, 205, 0]],
+                [[0, 0, 205]],
+                [[205, 205, 0]],
+                [[205, 0, 205]],
+                [[0, 205, 205]],
+            ],
+            dtype=np.uint8,
+        ),
+    )
+    light_scaling: float = kwargs.pop("light_scaling", 1.5)
+    colors_light: np.ndarray = np.trunc(light_scaling * colors)
+    np.clip(colors_light, 0, 255, out=colors_light)
+    colors: np.ndarray = np.hstack((colors, colors_light))
+
+    # Assert that there are a sufficient amount of colors
+    # and that all of the segmentations are of equal size
+    assert len(colors[:, 0, 0]) >= N
```

Comment:
```suggestion
    assert colors.shape[0] >= N
```

Why not "=="?

Reply:
I gave 6 colors in the default, and it doesn't really matter if we have more colors that unique components as only the first ones will be used anyways.

Reply:
Seems a bit random but OK for now. I think providing default list of standard colors (green, red, etc.) and extracting N many of them would feel like the most natural approach. Or one could utilize asuitable  linear colormap from matplotlib and simply consider N many equally distanced colors.

Reply:
That was a much better idea, and I fixed it now.